src/stream_base-inl.h

```diff
@@ -77,8 +77,14 @@ void StreamBase::AddMethods(Environment* env,
 template <class Base>
 void StreamBase::GetFD(Local<String> key,
                        const PropertyCallbackInfo<Value>& args) {
-  StreamBase* wrap = Unwrap<Base>(args.Holder());
+  Base* handle;
 
+  // Mimic implementation of StreamBase::GetFD() and UDPWrap::GetFD().
+  ASSIGN_OR_RETURN_UNWRAP(&handle,
+                          args.Holder(),
+                          args.GetReturnValue().Set(UV_EINVAL));
+
+  StreamBase* wrap = static_cast<StreamBase*>(handle);
```

Comment:
It definitely didn't work when I converted the returned `void*` to `StreamBase`. Hence why I first case to `Base`. I think this replicates what was happening before with `StreamBase* wrap = Unwrap<Base>()`. If you can think of a better way to ensure it casts properly I'll be happy to change it.

Duly notes. :)